src/sentry/static/sentry/app/views/alerts/rules/details/metricChart.tsx

```diff
@@ -1,145 +1,250 @@
+  state = {
+    width: -1,
+    height: -1,
   };
-  series.push(resolvedArea);
-  if (incidents) {
-    // select incidents that fall within the graph range
-    const periodStart = moment.utc(firstPoint);
-    const filteredIncidents = incidents.filter(incident => {
-      return !incident.dateClosed || moment(incident.dateClosed).isAfter(periodStart);
-    });
-
-    const criticalLines = filteredIncidents.map(incident => {
-      const detectTime = Math.max(moment(incident.dateStarted).valueOf(), firstPoint);
-      let resolveTime;
-      if (incident.dateClosed) {
-        resolveTime = moment(incident.dateClosed).valueOf();
-      } else {
-        resolveTime = lastPoint;
-      }
-      const line = [{coord: [detectTime, 0]}, {coord: [resolveTime, 0]}];
-      return line;
-    });
-    const criticalArea = {
-      seriesName: 'Critical Area',
-      type: 'line',
-      markLine: MarkLine({
-        silent: true,
-        lineStyle: {color: theme.red300, type: 'solid', width: 4},
-        data: criticalLines as any,
-      }),
-      data: [],
-    };
-    series.push(criticalArea);
-
-    const incidentValueMap: Record<number, string> = {};
-    const incidentLines = filteredIncidents.map(({dateStarted, identifier}) => {
-      const incidentStart = moment(dateStarted).valueOf();
-      incidentValueMap[incidentStart] = identifier;
-      return {xAxis: incidentStart};
-    });
-    const incidentLinesSeries = {
-      seriesName: 'Incident Line',
+
+  ref: null | ReactEchartsRef = null;
+
+  /**
+   * Syncs component state with the chart's width/heights
+   */
+  updateDimensions = () => {
+    const chartRef = this.ref?.getEchartsInstance?.();
+    if (!chartRef) {
+      return;
+    }
+
+    const width = chartRef.getWidth();
+    const height = chartRef.getHeight();
+    if (width !== this.state.width || height !== this.state.height) {
+      this.setState({
+        width,
+        height,
+      });
+    }
+  };
+
+  handleRef = (ref: ReactEchartsRef): void => {
+    // When chart initially renders, we want to update state with its width, as well as initialize starting
+    // locations (on y axis) for the draggable lines
+    if (ref && !this.ref) {
+      this.ref = ref;
+      this.updateDimensions();
+    }
+
+    if (!ref) {
+      this.ref = null;
+    }
+  };
+
+  getRuleCreatedThresholdElements = () => {
+    const {height, width} = this.state;
+    const {data, rule} = this.props;
+
+    if (!data.length || !data[0].data.length) {
+      return [];
+    }
+
+    const seriesData = data[0].data;
+    const seriesStart = seriesData[0].name as number;
+    const seriesEnd = seriesData[seriesData.length - 1].name as number;
+    const ruleChanged = moment(rule?.dateModified).valueOf();
+
+    const chartWidth = width - X_AXIS_BOUNDARY_GAP;
+    const position =
+      X_AXIS_BOUNDARY_GAP +
+      Math.round((chartWidth * (ruleChanged - seriesStart)) / (seriesEnd - seriesStart));
```

Comment:
nit: if position is a negative value we probably don't need to plot it as the dateModified wouldn't be in the range of the current viewing time period

Reply:
True we can return early when `ruleChanged < seriesStart`